### ISSUE TITLE
Material UI?

### ISSUE THREAD

**Issue by Jamdan2** [JetBrains/kotlin-wrappers#136]
Given that Material-UI is the most popular material design library for react, it might be nice to add an official wrapper. I'm happy to work on this. Would this repository be an appropriate place of this wrapper?

**Comment by Leonya** [comment]
Sure, go ahead!

**Comment by NorbertSandor** [comment]
@Jamdan2 
Where may I check the progress? Thanks.

**Comment by Jamdan2** [comment]
Haha havn't started this yet, though I will when I get the change. You can start it if you want, though!

**Comment by jpink** [comment]
There is few good starting points. Maybe we should ask them to join and make a pull request.

- [Kotlin Wrapper Library of Material-UI](https://github.com/subroh0508/kotlin-material-ui) from @subroh0508
  - That library uses quite old version 3.9, but seems to work on the newest one. I only have to set JVM max heap (using JAVA_OPTS) to get it build.
- [Muirwik - a Material UI React wrapper written in Kotlin](https://github.com/cfnz/muirwik) from @cfnz
  - That have newer version 4.4 and it needs refactoring enums to strings to get working on latest version 4.8. I got this also working. There is different approach on component creation because it doesn't use `ReactComponentBuilder`.
- [Kotlin wrapper for RMWC (React Material Web Components)](https://github.com/rivasdiaz/kotlin-rmwc) from @rivasdiaz
  - I have not tested. Seems to have only few wrappers.

We all have the same goal. Can we all (7) join and make this together? Ping @subroh0508, @cfnz, @rivasdiaz, @Jamdan2, @Leonya, @NorbertSandor?

**Comment by cfnz** [comment]
Happy to join/contribute. Only found @subroh0508 project a couple of days ago...

My project (Muirwik) started off as a learning exercise as described in the project's readme, but yes, happy to join/contribute.

**Comment by subroh0508** [comment]
Thank you for finding me! And happy to join/contribute, too!

**Comment by NikkyAI** [comment]
until there is a official wrapper, which library is currently working the best and compatible with kotlin-wrappers? i tried to use muirwik, but it has problems finding transitive dependnecies of material-ui (i used the 1.3.70-eap-42)

some stopgap solution would be nice, to put in `mavenLocal`

PS: will create a issue for muirwik later

**Comment by subroh0508** [comment]
> That library uses quite old version 3.9

I created a branch for updating Material-UI v4.x just before!
I'm going to migrate v3 -> v4 little by little.

https://github.com/subroh0508/kotlin-material-ui/tree/update-material-ui-v4

**Comment by NikkyAI** [comment]
i forked muirwik, changed the buildscript to multiplatform..
disabled the testapp (will eventually add that back)

https://github.com/NikkyAI/muirwik

i also added one little thing to `mButton`, it did not allow setting `type="submit"`

**Comment by cfnz** [comment]
Sounds good, I have been meaning to try out the newer buildscript functionality... shall check it out :-)

**Comment by NikkyAI** [comment]
there is still too many drawbacks for those buildscripts though.. for example no way to use devDependencies really
and the kotlin.js plugin could not create a artifact to publish on mavenLocal..
and then it turns out that depending on a artifact in mavenLocal made gradle crash for me on linux..
so i added it as git submodule / gradle subproject

you can find me in the kotlinlang slack btw, maybe the #react channel ?

**Comment by FrancoSabadini** [comment]
Hi @jpink, @subroh0508, @cfnz, @rivasdiaz, @Jamdan2, @Leonya, @NorbertSandor did you guys start working on this together already? If so, which repo of all the ones listed is the one you guys are supporting?

**Comment by NikkyAI** [comment]
i am still using muirwik with the buildscript replaced (with kotlin-multiplatform)

**Comment by subroh0508** [comment]
I've just finished the task migrating Material-UI v3 -> v4 👍 

Please use if you like 😉

https://github.com/subroh0508/kotlin-material-ui/pull/9

**Comment by cfnz** [comment]
I have done some minor updates to muirwik but have not had a chance to look at merging with/supporting other projects yet... Would be nice to have one project to rule (and be better than) them all, but no time to look into it (yet?)...

**Comment by krzema12** [comment]
How about (semi)-automatically generating these wrappers? Material UI comes with TypeScript typings, so in theory one could parse them to generate Kotlin typings. Seems like a more scalable way than crafting these manually.

I also thought about using https://github.com/Kotlin/dukat which is being created to serve this purpose. I gave it a naive shot and first thing I noticed is that the strings which have well-defined subsets in TypeScript are mapped to plain strings in Kotlin. I didn't look further what's missing.

Apparently someone has already tried Dukat with Material UI. The project has these issues open:
* https://github.com/Kotlin/dukat/issues/162
* https://github.com/Kotlin/dukat/issues/250
